### DISCUSSION TITLE
brpc协议识别效率问题疑惑

### DISCUSSION THREAD

**Discussion by zcl1115** [harmonycloud/kindling#603] [agent collector] [answered]
仿造dubbo2协议开发brpc协议的分析器，但是看到官方文档请求的元数据信息（服务名、方法名、错误以及请求响应）需要使用protobuf反序列化，想问一下引入这种会对kindling本身的效率有影响或者对agent有负担吗
<img width="589" alt="image" src="https://github.com/KindlingProject/kindling/assets/45005291/fb79168e-0e50-408a-83f4-17e6edd033ab">

**Comment by dxsup** [comment, marked as answer]
反序列化protobuf，肯定要比只进行字符对比判断的计算量更大，所以会提高CPU使用量。这可能会导致事件处理效率下降，意味着能够及时处理的事件数下降，进而影响能够分析的请求量。

不过这个解析是用户态的，肯定不会影响应用本身。建议通过实际测试观察对agent的影响有多大。
